Logging

emdete opened the pull request:
implement some of the ideas of https://github.com/qtile/qtile/issues/841

flacjacket commented:
Other than those couple things, you need to rebase this on top of the current develop branch, as it won't apply cleanly.

emdete commented:
about INFO: this was an error as i prefere INFO, still i assume that most of the messages should go to DEBUG and INFO should be the default level.

about rebase: sorry, i did not pull before, this is now fixed.

about configurability: i will take a look to properly solve this. for now command line is the only option, the other stuff probably never worked anyway. are you OK with this?

emdete commented:
i struggled a bit over the declaration of the coding of the sources which is set to utf-8 in only some sources (mainly to support the names in the copyright area i assume) and the setting for vim to follow PEP8. because i like the vim settings (i struggled with 4-spaces in some of my commits) i added those to all sources as first line (because i found that in some sources this way) while i know this setting in the last line. anyway i seem to have destroyed the coding settings which i now reverted and put the coding: into the first line, vim in the second. i hope travis and the developers like it this way.

flacjacket commented:
I'll take one last look through this.

I don't really mind the vim tabstop declarations. I just have that in my vimrc, but it is the convention we use, so I'd say it's ok.

flacjacket commented on libqtile/window.py:
```diff
@@ -1,3 +1,5 @@
+# vim: tabstop=4 shiftwidth=4 expandtab
+# vim: tabstop=4 shiftwidth=4 expandtab
```
Line repeated

flacjacket commented on libqtile/xcbq.py:
```diff
@@ -1,3 +1,5 @@
+# vim: tabstop=4 shiftwidth=4 expandtab
+# vim: tabstop=4 shiftwidth=4 expandtab
```
Also here

flacjacket commented on libqtile/xcursors.py:
```diff
@@ -1,4 +1,7 @@
+# vim: tabstop=4 shiftwidth=4 expandtab
+# vim: tabstop=4 shiftwidth=4 expandtab
```
And here

flacjacket commented on docs/conf.py:
```diff
@@ -1,3 +1,4 @@
+# vim: tabstop=4 shiftwidth=4 expandtab
```
Put below utf-8 line

ramnes replied:
Yes, or just configure your editor rather than doing that.

emdete commented:
i hope you give it another look when i have a solution for configurable logging 😉

i will develop something to allow reconfiguration from the config.py but had no time till now. pure command line configuration would be easier but not desired as far as i understood, right?

flacjacket commented:
Oh, I'll definitely keep giving feedback, I was just thinking I would take another look at it and be able to merge it :) There are the couple things I mentioned that I found that should be fixed, but otherwise I'd be happy to merge this as is. The log configuration is likely broken right now I don't mind getting rid of it for now. Command line options would be ok for a start, but you don't always have easy access to command line option when starting a window manager through a login manager, so access to configuration through configuration files is nice to add.

ramnes commented on bin/qtile:
```diff
@@ -56,13 +59,13 @@ def make_qtile():
-        prog='qtile')
-
+        prog='qtile',
+        )
```
Why changing the indentation and adding an useless comma?

emdete replied:
i use that pattern to be able to easier

 - sort parameters

 - add more parameters or comment them

without struggling over missing commas or brackets.

ramnes replied:
That's ugly.

ramnes commented on docs/sphinx_qtile.py:
```diff
@@ -1,3 +1,4 @@
+# vim: tabstop=4 shiftwidth=4 expandtab
```
Please configure your editor rather than doing that.

Roger commented:
about the vim config line, we can use something like this on the root of the repo: http://editorconfig.org

flacjacket commented:
From #841, you can drop the `getLogger()` lines for either importing a single logger or using the root logger. If you can drop the vim lines and whitespace changes, this will be good.
